### PR TITLE
[azure-core-cpp] Don't refer to versions of curl that don't exist.

### PR DESCRIPTION
**Pull request by BillyONeal**
There's a "version>=" against a version of curl that does not exist in the version database. This causes customers that use versioning with Azure components to fail to resolve a version; versions need to be in the version database for us to be able to compare them.

### CODE REVIEW

**Comment by dg0yt** [comment]
Isn't this a bug in the tool? Why does it need one `=` match when there is an unlimited number of  `>` matches? 
You may not want to remove versions from the vcpkg registry, but do you want to restrict what users do in their registries?

**Review by Adela0814 — APPROVED** [review]

**Comment by BillyONeal** [comment]
> Isn't this a bug in the tool?

No.

> Why does it need one `=` match when there is an unlimited number of `>` matches? You may not want to remove versions from the vcpkg registry, but do you want to restrict what users do in their registries?

There is no "`>` match". The tool doesn't know what "7.44" is. Therefore, it doesn't know what version scheme it uses. Therefore it can't be compared to anything else. (If we took the conservative outcome and interpreted as `version-string`, this situation would still hose customers, because `version-string` is similarly unorderable)

The limitation that the version database has to know about all versions compared with is the tradeoff for

1. not needing to repeat the version scheme you want to use in all the places, and
2. supporting different version schemes in the first place

For example, `version-date` and `version-string` can recognize the same input but choose different sort orders.

I agree that the situation is improvable, for example by allowing the version database to contain versions that can't actually be 'hydrated' so that they can get a scheme, but that enters new feature territory rather than bug territory.

**Comment by ahsonkhan** [comment]
In that case, do we (or can we) have tooling to detect when a change is pointing to a non-existing version and throw an error, so the registry author knows that something isn't correct?

File: versions/a-/azure-core-cpp.json [hunk 1]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "bfb8b85899d4e6783f93392104d7ee80c8d0a89c",
+      "version-semver": "1.10.2",
+      "port-version": 1
+    },
```
Comment (ahsonkhan) [inline versions/a-/azure-core-cpp.json:7]: I am trying to understand the fix here. Why was this change needed? What's the difference between this commit hash and the one below with the same version-semver, but different port-version?

**Comment by ahsonkhan** [comment]
Do we prevent removal of versions/ports from the registry? What happens if the curl package author decide to remove version 7.48 from https://github.com/microsoft/vcpkg/blob/master/versions/c-/curl.json#L425? Would upstream packages like azure-core-cpp be broken?

**Comment by teo-tsirpanis** [comment]
> What happens if the curl package author decide to remove version 7.48

Versions are not removed.

**Comment by teo-tsirpanis** [comment]
> Even if the port author removes the version?

Guess even then, but I'm not sure.
